src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -397,16 +397,18 @@ private static boolean isNonLocalTypeDeclaration(DetailAST typeDeclAst) {
     private static DetailAST getBlockContainingLocalAnonInnerClass(DetailAST literalNewAst) {
         DetailAST currentAst = literalNewAst;
         DetailAST result = null;
-        while (!TokenUtil.isOfType(currentAst, CONTAINERS_FOR_ANON_INNERS)) {
-            if (currentAst.getType() == TokenTypes.LAMBDA
-                    && currentAst.getParent()
-                    .getParent().getParent().getType() == TokenTypes.OBJBLOCK) {
-                result = currentAst;
-                break;
+        DetailAST topMostLambdaAst = null;
+        while (currentAst != null && !TokenUtil.isOfType(currentAst, CONTAINERS_FOR_ANON_INNERS)) {
+            if (currentAst.getType() == TokenTypes.LAMBDA) {
+                topMostLambdaAst = currentAst;
```

Comment:
This is a time-complexity mutation. The reason I have added this to the suppressions list is because removing this condition will make the entire `COMPILATION_UNIT` to be processed again. And having many nested lambdas can make the entire `COMPILATION_UNIT` be processed multiple times.

If the penalty was small, I would have inclined towards pitest, but processing the entire `COMPILATION_UNIT` multiple times isn't a wise option IMO.